packages/labs/task/src/test/task_test.ts

```diff
@@ -273,4 +273,41 @@ suite('Task', () => {
     await tasksUpdateComplete();
     assert.equal(el.renderedStatus, 'result: zot3');
   });
+
+  test('task functions can return initial state', async () => {
+    class TestEl extends ReactiveElement {
+      @property()
+      state = '';
+
+      task = new Task(
+        this,
+        async ([state]) => (state === 'initial' ? initialState : 'A'),
+        () => [this.state]
+      );
+
+      t2 = new Task(
+        this,
+        () => new Promise(() => []),
+        () => []
+      );
+    }
+    customElements.define(generateElementName(), TestEl);
+
+    const el = new TestEl();
+    assert.equal(el.task.status, TaskStatus.INITIAL, 'initial');
+    container.append(el);
+
+    // After one microtask we expect the task function to have been
+    // called, but not completed
+    await Promise.resolve();
```

Comment:
Make the task resolution explicit so we're not depending on this?

Reply:
Added a `taskComplete` promise, but that doesn't address this await which is waiting for the task function to be called, but not completed. It did replace the next `await Promise.resolve()` though.